azkaban-common/src/main/java/azkaban/version/JdbcVersionSetLoader.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.version;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A JDBC based implementation for loading VersionSet from the backed db. This class keeps the copy
+ * of entries in the version_set table locally as well to reduce the number of queries to the db.
+ * <p>
+ * This is a singleton class to be instantiated using dependency injection.
+ */
+@Singleton
+public class JdbcVersionSetLoader implements VersionSetLoader {
```

Comment:
This class can be renamed to something like VersionSetLoaderImpl or VersionSetDaoImpl. So it would make sense to move this peace of code to cf-engine-k8s branch so that it can be place under imagemgmt package and similar structure and format can be followed.

Reply:
It does make sense to change the package to imagemgmt, however, since the code itself doesn't have any dependency with the code in cf-engine-k8s, it can be kept in the master branch itself.

Reply:
Since it exetends VersionSetLoader, changing it to VersionSetDaoImpl would be odd. This name is derived from the pair FlowTriggerInstanceLoader and JdbcFlowTriggerInstanceLoaderImpl already existing in the code base.